C/Coin-OR/Osi/build_tarballs.jl

```diff
@@ -53,11 +53,21 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("CoinUtils_jll", CoinUtils_version),
+    Dependency("CoinUtils_jll", CoinUtils_version, compat="$(CoinUtils_version)"),
```

Comment:
Is adding these sort of compat bounds a good idea?

Reply:
I'd say yes. Is that a plain upstream version or the one with the offsets? Note that if you specify only the compat version, the minimum version is used also as build version, so you can remove the second argument

Reply:
It's one with the offsets. (I'm going to make all the COIN-OR stuff have these offsets. When building out the original tree we had to deal with some inconsistencies between upstream version and the JLL. Never again.)

So If we want to bump the upstream patch release from `2.11.4` to `2.11.5`, but it breaks in some way, we should be able to bump the offset from `0.0.0` to `1.0.0` and it will go from `200.1100.400` to `201.1100.500`.

Reply:
Ok, the offset are a bit tricky because you may report as compatible other interleaved versions with different offsets. @fingolfin or @benlorenz may have more experience about that

Reply:
I think it should be okay? It declares that it supports any version `200.1100.400` or later.

 * If upstream patch changes, it becomes `200.1100.500`  -> compatible
 * If upstream minor changes, it becomes `200.1200.0` -> compatible
 * If upstream major changes, it becomes `300.0.0` -> incompatible
 * If offset patch changes, it becomes `200.1100.401`  -> compatible
 * If offset minor changes, it becomes `200.1101.400`  -> compatible
 * If offset major changes, it becomes `201.1100.400` -> incompatible

Some future upstream like `2.14.3` at offset `0.4.2` becomes `200.1404.302` so we're still good. When the upstream version is changed the offset resets to `0.0.0`.